liquidator/liquidator.js

```diff
@@ -249,50 +269,129 @@ class Liquidator {
       );
 
       // Simple version of inventory management: simulate the transaction and assume that if it fails, the caller didn't have enough collateral.
-      let liquidationResult, gasEstimation;
-      try {
-        [liquidationResult, gasEstimation] = await Promise.all([
-          liquidation.call({ from: this.account }),
-          liquidation.estimateGas({ from: this.account })
+      const notEnoughCollateral = await liquidation
+        .call({ from: this.account })
+        .then(() => false)
+        .catch(() => true);
+
+      // Attempts to swap some capital for tokenCurrency
+      // Mutates the state of notEnoughCollateral in this logical scope
+      if (notEnoughCollateral) {
+        const tokenCurrency = await this.empClient.getTokenCurrency();
+
+        // reserveWeiNeeded is a reverse calculation to estimate how much capital we need to get the amount of `tokensToLiquidate`
+        // While oneGweiReturn is used as a reference point to determine slippage
+        const [reserveWeiNeeded, oneGweiReturn] = await Promise.all([
+          this.oneInch.getExpectedReturn({
+            fromToken: tokenCurrency.address,
+            toToken: this.reserveCurrencyAddress,
+            amountWei: tokensToLiquidate.toString()
+          }),
+          this.oneInch.getExpectedReturn({
+            fromToken: tokenCurrency.address,
+            toToken: this.reserveCurrencyAddress,
+            amountWei: this.toWei("1", "gwei")
+          })
         ]);
-      } catch (error) {
-        this.logger.error({
+
+        this.logger.info({
           at: "Liquidator",
-          message:
-            "Failed to liquidate position: not enough synthetic (or large enough approval) to initiate liquidation❌",
-          sponsor: position.sponsor,
-          inputPrice: scaledPrice.toString(),
-          position: position,
-          minLiquidationPrice: this.liquidationMinPrice,
-          maxLiquidationPrice: maxCollateralPerToken.toString(),
-          tokensToLiquidate: tokensToLiquidate.toString(),
-          error
+          message: `Attempting to convert reserve currency ${this.reserveCurrencyAddress} to tokenCurrency ${tokenCurrency.address} ❌`
```

Comment:
If this is no longer an error then perhaps remove the emoji? Also can you add more information to this log? there is a lot of stuff you can add here.

Also nowhere else do we supplement the messages like this; rather put `reserveCurrencyAddress` and `tokenCurrency.address` as properties in the logs, the same way it's done elsewhere. If you could add in the `oneGweiReturn`, `reserveWeiNeeded` and any other information about the swal like `tokensToLiquidate` it would be more useful than this log.

Reply:
Seems like the result from auto merging, will fix it

Reply:
@kendrickta you resolved this issue but I don't think it's resolved. My comments are still relevant: 
1) this should not be a `info` log, it should be a debug.
2) you should not write the messages like this with inline info. we don't do this anywhere else in the repo.
3) you should add additional information to the log message with pretty much all state at that point.
4) you should not have an emoji.

This log is similar to the one [here](https://github.com/UMAprotocol/protocol/blob/8aae10fee7c944351f792cf72c924899ad9fa216/packages/liquidator/src/liquidator.js#L281) in the bots that happen _before_ some async action.